pkg/wireguard/agent/agent.go

```diff
@@ -48,6 +46,7 @@ import (
 )
 
 var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "wireguard")
+var wgDummyPeerKey = wgtypes.Key{}
```

Comment:
How does this look in `wg show`? I also wonder if we want to validate `UpsertPeer` to make sure no malicious CilumNode ever claims to be the dummy peer (which would probably break things).

Reply:
> I also wonder if we want to validate UpsertPeer to make sure no malicious CilumNode ever claims to be the dummy peer (which would probably break things).

I added a check in `UpdatePeer()` to make sure a node can't claim the dummy peer key.